src/OpenLoco/src/Vehicles/VehicleHead.cpp

```diff
@@ -4198,12 +4200,184 @@ namespace OpenLoco::Vehicles
         });
     }
 
+    template<typename T>
+    static void setTrainModFlags(Vehicle& train, T& trackRoadObj)
+    {
+        bool canRoll = true;
+        uint32_t roadMods = 0;
+        uint32_t rackRailMods = 0;
+        for (auto& car : train.cars)
+        {
+            auto* vehicleObj = ObjectManager::get<VehicleObject>(car.front->objectId);
+            if (vehicleObj->bodySprites[0].numRollFrames == 1)
+            {
+                canRoll = false;
+            }
+            if (vehicleObj->hasFlags(VehicleObjectFlags::rackRail))
+            {
+                for (auto i = 0U; i < trackRoadObj.numMods; ++i)
+                {
+                    if (trackRoadObj.mods[i] == vehicleObj->rackRailType)
+                    {
+                        rackRailMods |= (1U << i);
+                        break;
+                    }
+                }
+            }
+
+            for (auto i = 0U; i < vehicleObj->numTrackExtras; ++i)
+            {
+                const auto mod = vehicleObj->requiredTrackExtras[i];
+                for (auto j = 0U; j < trackRoadObj.numMods; ++j)
+                {
+                    if (trackRoadObj.mods[j] == mod)
+                    {
+                        roadMods |= (1U << j);
+                        break;
+                    }
+                }
+            }
+        }
+        train.head->var_53 = roadMods;
+        train.veh1->var_49 = rackRailMods;
+        train.head->var_38 &= ~Flags38::fasterAroundCurves;
+        if (canRoll)
+        {
+            train.head->var_38 |= Flags38::fasterAroundCurves;
+        }
+    }
+
     // 0x004B7CC3
     void VehicleHead::sub_4B7CC3()
```

Comment:
Perhaps something as simple as `updateVehicleProperties` or even `updateProperties`? Either would be better than just `sub_4B7CC3`, I think.